plugins/modules/katello_lifecycle_environment.py

```diff
@@ -70,96 +73,43 @@
-            module.fail_json(msg="Prior cannot be updated on a lifecycle environment.")
-        desired_environment = LifecycleEnvironment(name=name, organization=organization, label=label, description=description, prior=prior)
-        fields = ['description']
-        if current_environment is not None:
-            (needs_update, le) = update_fields(desired_environment, current_environment, fields)
-            if needs_update:
-                if not module.check_mode:
-                    le.update(fields)
-                changed = True
-        else:
-            desired_environment.prior = find_prior(module, "Library", organization) if prior is None else prior
-            if not module.check_mode:
-                desired_environment.create()
-            changed = True
-    elif current_environment is not None:
-        if not module.check_mode:
-            current_environment.delete()
-        changed = True
-    return changed
+from ansible.module_utils.foreman_helper import KatelloEntityApypieAnsibleModule
 
 
 def main():
-    module = KatelloEntityAnsibleModule(
-        argument_spec=dict(
+    module = KatelloEntityApypieAnsibleModule(
+        entity_spec=dict(
             name=dict(required=True),
             label=dict(),
             description=dict(),
-            prior=dict(),
+            prior=dict(type='entity', flat_name='prior_id'),
         ),
     )
 
-    (module_params, state) = module.parse_params()
-    name = module_params.get('name')
-    label = module_params.get('label') if module_params.get('label') != '' else None
-    description = module_params.get('description')
-    prior = None if module_params.get('prior') == '' else module_params.get('prior')
-    organization = module_params.get('organization')
+    entity_dict = module.clean_params()
 
     module.connect()
 
-    validate_params(module, state, label=label, description=description, prior=prior)
+    entity_dict['organization'] = module.find_resource_by_name('organizations', entity_dict['organization'], thin=True)
+    scope = {'organization_id': entity_dict['organization']['id']}
+    entity = module.find_resource_by_name('lifecycle_environments', name=entity_dict['name'], params=scope, failsafe=True)
+    if not module.desired_absent:
+        if 'prior' in entity_dict:
+            entity_dict['prior'] = module.find_resource_by_name('lifecycle_environments', entity_dict['prior'], params=scope, thin=True)
+        # Default to 'Library' for new env with no 'prior' provided
+        elif not entity:
+            entity_dict['prior'] = module.find_resource_by_name('lifecycle_environments', 'Library', params=scope, thin=True)
+
+    if entity:
+        if 'label' in entity_dict and entity_dict['label'] and entity['label'] != entity_dict['label']:
+            module.fail_json(msg="Label cannot be updated on a lifecycle environment. %s ******* %s" % (entity_dict, entity))
```

Comment:
why do you print the two entities with `%s ******* %s` here?

Reply:
:man_facepalming: